mslib/mscolab/server.py

```diff
@@ -814,12 +802,10 @@ def idp_login():
         _, response_binding = sp_config.config.getattr("endpoints", "sp")[
             "assertion_consumer_service"
         ][0]
-        relay_state = rndstr()
         entity_id = get_idp_entity_id(selected_idp)
         _, binding, http_args = sp_config.prepare_for_negotiated_authenticate(
             entityid=entity_id,
             response_binding=response_binding,
-            relay_state=relay_state,
```

Comment:
I am not sure what the implications of removing relay_state completely are. pysaml2 seems to have a default value of the empty string for that parameter. This value is internally (in pysaml2) also referred to as `session_id`, which makes it sound important. Then again, naively I would assume that if it has a default value it *should* be safe to omit.

Reply:
The `relay_state` parameter in requests is optional and serves to maintain the context of an application during SSO. It lets the SP specify where to redirect the user after SSO, such as a particular page or resource. When the user returns to the SP after authentication, the SP can use `relay_state` from the SAML response to decide where to send the user next, ensuring a seamless user experience. By default, as you said in [pysaml2, relay_state](https://github.com/IdentityPython/pysaml2/blob/cae1326ae05bf8039c5990b13ea46b51a1a92aa5/src/saml2/client.py#L113) is often an empty string, but the SP can customize it to convey specific information or instructions as needed. 

I believe that, in our case, since we are navigating from msui to mscolab, the authentication is the sole purpose. Therefore, in this scenario, there's no need to maintain the context of an mscolab during SSO. By the way, I truly appreciate your valuable insights.

Reply:
Right, thanks for the clarification, that makes sense. The parameter can be omitted then, in our case.